File: Tutorials/StylingExample/netcore/StyleFactory.cs
using System.Drawing;
using SpaceVIL;
using SpaceVIL.Common;
using SpaceVIL.Core;
using SpaceVIL.Decorations;

namespace StylingExample
{
    public static class StyleFactory
    {
        public static Style GetPictureStyle()
        {
            // style for Picture
            Style style = new Style();
            style.SetSizePolicy(SizePolicy.Expand, SizePolicy.Expand);
            style.SetSpacing(0, 5);
            style.SetMargin(3, 3, 3, 3);
            style.SetPadding(5, 5, 5, 5);
            style.SetAlignment(ItemAlignment.Left, ItemAlignment.Top);
            style.SetBackground(80, 80, 80);
            style.SetShadow(new Shadow(5, 0, 0, Color.FromArgb(200, 0, 0, 0)));
            style.IsShadowDrop = true;
            style.SetBorder(Color.Gray, new CornerRadius(10), 1);

            // inner stylesExpand: Picture consist of ImageItem and Label
            // stylHCentere for ImageItemVCenter
            Style imageStyle = new Style();
            imageStyle.SetSizePolicy(SizePolicy.Expand, SizePolicy.Expand);
            imageStyle.SetAlignment(ItemAlignment.HCenter, ItemAlignment.VCenter);
            imageStyle.SetBackground(45, 45, 45);
            imageStyle.BorderRadius = new CornerRadius(6, 6, 0, 0);
            style.AddInnerStyle("image", imageStyle);
            // style for Label
            Style textStyle = new Style();
            textStyle.SetSizePolicy(SizePolicy.Expand, SizePolicy.Fixed);
            textStyle.Height = 30;
            textStyle.SetTextAlignment(ItemAlignment.HCenter, ItemAlignment.VCenter);
            // textStyle.SetAlignment(ItemAlignment.LEFT, ItemAlignment.BOTTOM);
            textStyle.SetBackground(150, 150, 150);
            textStyle.SetForeground(32, 32, 32);
            textStyle.BorderRadius = new CornerRadius(0, 0, 6, 6);
            style.AddInnerStyle("text", textStyle);
            return style;
        }

        public static void UpdateWrapGridStyle()
        {
            // get style from basic theme
            Style selectionStyle =
                DefaultsService.GetDefaultTheme().GetThemeStyle(typeof(WrapGrid)).GetInnerStyle("area").GetInnerStyle("selection");
            // change style for selection
            if (selectionStyle != null)
            {
                selectionStyle.AddItemState(ItemStateType.Hovered, new ItemState(Color.FromArgb(10, 162, 232)));
                selectionStyle.AddItemState(ItemStateType.Toggled, new ItemState(Color.FromArgb(49, 213, 121)));
                selectionStyle.BorderRadius = new CornerRadius(12);
                selectionStyle.SetPadding(0, 0, 0, 0);
            }
        }
    }
}

File: Tutorials/StylingExample/jvm/src/main/java/StylingExample/StyleFactory.java
package StylingExample;

import java.awt.Color;

import com.spvessel.spacevil.WrapGrid;
import com.spvessel.spacevil.Common.DefaultsService;
import com.spvessel.spacevil.Decorations.CornerRadius;
import com.spvessel.spacevil.Decorations.ItemState;
import com.spvessel.spacevil.Decorations.Shadow;
import com.spvessel.spacevil.Decorations.Style;
import com.spvessel.spacevil.Flags.ItemAlignment;
import com.spvessel.spacevil.Flags.ItemStateType;
import com.spvessel.spacevil.Flags.SizePolicy;

public final class StyleFactory {

    private StyleFactory() {
    }

    public static Style getPictureStyle() {
        // style for Picture
        Style style = new Style();
        style.setSizePolicy(SizePolicy.EXPAND, SizePolicy.EXPAND);
        style.setSpacing(0, 5);
        style.setMargin(3, 3, 3, 3);
        style.setPadding(5, 5, 5, 5);
        style.setAlignment(ItemAlignment.LEFT, ItemAlignment.TOP);
        style.setBackground(80, 80, 80);
        style.setShadow(new Shadow(5, 0, 0, new Color(0, 0, 0, 200)));
        style.isShadowDrop = true;
        style.setBorder(Color.gray, new CornerRadius(10), 1);

        // inner styles: Picture consist of ImageItem and Label
        // style for ImageItem
        Style imageStyle = new Style();
        imageStyle.setSizePolicy(SizePolicy.EXPAND, SizePolicy.EXPAND);
        imageStyle.setAlignment(ItemAlignment.HCENTER, ItemAlignment.VCENTER);
        imageStyle.setBackground(45, 45, 45);
        imageStyle.borderRadius = new CornerRadius(6, 6, 0, 0);
        style.addInnerStyle("image", imageStyle);
        // style for Label
        Style textStyle = new Style();
        textStyle.setSizePolicy(SizePolicy.EXPAND, SizePolicy.FIXED);
        textStyle.height = 30;
        textStyle.setTextAlignment(ItemAlignment.HCENTER, ItemAlignment.VCENTER);
        textStyle.setBackground(150, 150, 150);
        textStyle.setForeground(32, 32, 32);
        textStyle.borderRadius = new CornerRadius(0, 0, 6, 6);
        style.addInnerStyle("text", textStyle);
        
        return style;
    }

    public static void updateWrapGridStyle() {
        // get style from basic theme
        Style selectionStyle = DefaultsService.getDefaultTheme().getThemeStyle(WrapGrid.class).getInnerStyle("area")
                .getInnerStyle("selection");
        // change style for selection
        if (selectionStyle != null) {
            selectionStyle.addItemState(ItemStateType.HOVERED, new ItemState(new Color(10, 162, 232)));
            selectionStyle.addItemState(ItemStateType.TOGGLED, new ItemState(new Color(49, 213, 121)));
            selectionStyle.borderRadius = new CornerRadius(12);
            selectionStyle.setPadding(0, 0, 0, 0);
        }
    }
}
